### PR TITLE
Rhmap 2577.remove.deprecated.libs

### PR DESCRIPTION
**Pull request by corinnekrych**
@wei-lee mind to review?
This PR is to remove ASIHttp dependency.
We also remove the option of having synchronous call.

### CODE REVIEW

File: fh-ios-sdk/FHHttpClient.m [hunk 1]
```diff
@@ -45,85 +71,64 @@ - (void)sendRequest:(FHAct *)fhact
...
-
-        if ([request responseStatusCode] == 200) {
-            [self successWithResponse:fhResponse AndAction:fhact];
+    NSMutableURLRequest* brequest = [self request:fhact withUrl:apicall];
+    NSURLRequest* request = [brequest copy];
+    brequest.timeoutInterval = fhact.requestTimeout;
+    
+    NSURLSessionDataTask * task = [_session dataTaskWithRequest:brequest completionHandler:^(NSData* data, NSURLResponse* response, NSError* error) {
+        NSString *encodingName = [response textEncodingName];
+        NSStringEncoding encodingType = NSASCIIStringEncoding;
```
Comment (wei-lee) [inline fh-ios-sdk/FHHttpClient.m]: probably better to use NSUTF8StringEncoding by default?

File: fh-ios-sdk/FHHttpClient.m [hunk 2]
```diff
@@ -45,85 +71,64 @@ - (void)sendRequest:(FHAct *)fhact
...
+        if (encodingName != nil) {
+            encodingType = CFStringConvertEncodingToNSStringEncoding(CFStringConvertIANACharSetNameToEncoding((CFStringRef)encodingName));
+        }
+        NSString* reponseAsRawString = [[NSString alloc] initWithBytes:[data bytes] length:[data length] encoding:encodingType];
+        int statusCode = (int)((NSHTTPURLResponse*)response).statusCode;
+        if (error) {
+            DLog(@"Error %@", error.description);
```
Comment (wei-lee) [inline fh-ios-sdk/FHHttpClient.m]: Should we move this into the dispatch_async block as well?

**Comment by wei-lee** [comment]
reviewed

**Comment by corinnekrych** [comment]
landed b24010adb8adbe06dad420e8b985ac2b67e3fd65

**Pull request closed without merging**